docs/reference/configuration/BLE.md

```diff
@@ -0,0 +1,52 @@
+<h2 id="configuration-ble">BLE</h2>
+
+This page describes build-time configurable parameters for BLE. A resource constrained device can limit its functionality to decrease the memory and flash usage of the BLE module. 
+
+To understand what each feature provides please refer to the API docs and the Bluetooth specification.
+
+#### Build time configuration of the BLE feature
+
+To minimise the size of the BLE stack, BLE defines a set of build options.
+
+The configuration is contained in the `mbed_lib.json` configuration file. By default all the features are enabled.
+
+Turning off individual features will remove the code and any memory allocations required by that feature. Some features depend on each other - see the comments in the configuration file. These dependencies will be enforced during compile time.
+
+Trying to use a feature which has been disabled will result in a compile time error or an error reporting the feature as unimplemented at runtime.
+
+#### Changing the configuration
+
+An excerpt from the configuration file:
+
+```
+{
+    "name": "ble",
+    "config": {
+        "ble-role-observer": {
+            "help": "Include observer BLE role support, allows listening for and processing advertising packets.",
+            "value": true,
+            "macro_name": "BLE_ROLE_OBSERVER"
+        },
+    ...
+```
+
+By changing `"value": true,` to `false` you can disable each feature.
+
+##### Configurable features
+
+These are feature that can be disabled:
+
+- observer (scanning)
```

Comment:
Would you be able to replace this with a table (config flag and description)?

Reply:
Where does the BLE mbed_lib.json file live? I was only able to find target specific ones and 
`mbed compile --config -v --prefix ble` doesn't give me anything (tried with a BLE target specified as well).

Reply:
@paul-szczepanek-arm Pinging for above question. If the table in this doc is a list of configuration options available through the config system then the doc config list auto-updater script won't catch it unless it's formatted like the other files:

https://github.com/ARMmbed/mbed-os-5-docs/pull/948

Reply:
Do I still need to write this if it's autogenerated? How do I insert the autogenerated part? May I have the script?